src/zcl_abapgit_factory.clas.abap

```diff
@@ -154,4 +170,32 @@ CLASS zcl_abapgit_factory IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD get_branch_overview.
+
+    DATA: ls_branch_overview LIKE LINE OF gt_branch_overview,
+          lv_repo_key        TYPE ty_branch_overview-repo_key.
+    FIELD-SYMBOLS: <ls_branch_overview> TYPE zcl_abapgit_factory=>ty_branch_overview.
+
+    lv_repo_key = io_repo->get_key( ).
+
+    READ TABLE gt_branch_overview ASSIGNING <ls_branch_overview>
+                               WITH TABLE KEY repo_key = lv_repo_key.
```

Comment:
this looks dangerous, the repo_key ids are reused, if a repo is deleted and another added, then it gets the same key, and io_repo is stored in the singleton.

Suggest not making this a singleton, instead use an internal table with injections for unit testing

Reply:
I corrected it in https://github.com/larshp/abapGit/pull/1665
The implementation of your suggestion I will do later.

Reply:
Ah, good to know. Then it wasn't the best idea.

Reply:
it is also not the best idea the way it finds the keys 😄 I just dont like singletons, they sometimes introduce evil bugs